sw/tests/bare-metal/hostd/idma_test.c

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include "util.h"
+#include "io.h"
+#include "printf.h"
+
+#define IDMA_BASE          0x01000000
+#define SRC_ADDR           0x14000000
+#define DST_ADDR           0x14003000
+
+#define IDMA_SRC_ADDR_OFFSET         0x000000d8
+#define IDMA_DST_ADDR_OFFSET         0x000000d0
+#define IDMA_LENGTH_OFFSET           0x000000e0
+#define IDMA_NEXT_ID_OFFSET          0x00000044
+#define IDMA_REPS_2                  0x000000f8
+#define IDMA_CONF                    0x00000000
```

Comment:
Would it make sense to add the dma regs header file like we did for the other IPs? Like [here](https://github.com/pulp-platform/astral/tree/astral/sw/include/regs)

Reply:
Worst case we could just copy the header from Cheshire to here. Chao added [the rule](https://github.com/pulp-platform/cheshire/blob/cl/idma-wrapper/cheshire.mk#L135) to generate those regs but it breaks due to [flatdict](https://github.com/pulp-platform/cheshire/blob/cl/idma-wrapper/requirements.txt#L10) package missing into the server. I asked for hints also to @alex96295